### PR TITLE
fix: show pending ban/unban/kick state if the control node is offline

### PR DESCRIPTION
**Pull request by mprakhov**
### What does the PR do

Show pending ban/unban/kick state if the control node is offline

Closes: https://github.com/status-im/status-desktop/issues/13596

### Affected areas

kick/ban/unban state in the member's list

### CODE REVIEW

File: src/app/modules/main/module.nim [hunk 1]
```diff
@@ -344,6 +344,29 @@ proc createChannelGroupItem[T](self: Module[T], channelGroup: ChannelGroupDto):
...
+        localNickname = contactDetails.dto.localNickname,
+        alias = contactDetails.dto.alias,
+        icon = contactDetails.icon,
+        colorId = contactDetails.colorId,
+        colorHash = contactDetails.colorHash,
+        onlineStatus = toOnlineStatus(self.controller.getStatusForContactWithId(memberId).statusType),
+        isContact = contactDetails.dto.isContact,
+        isVerified = contactDetails.dto.isContactVerified(),
+        membershipRequestState = state,
+      )
```
Comment (jrainville) [inline src/app/modules/main/module.nim:367]: Small suggestion, maybe we can create util function in this module where we pass the `contactDetails` and `memberId` and it does the `initMemberItem`.  That way we don't have to duplicate all those lines.

**Review by jrainville — APPROVED** [review]

File: src/app/modules/main/module.nim (hunk 1 above)
Comment (mprakhov) [reply src/app/modules/main/module.nim]: done 👍

File: src/app/modules/main/module.nim [hunk 2]
```diff
@@ -1649,4 +1596,23 @@ method checkIfAddressWasCopied*[T](self: Module[T], value: string) =
     return
   self.addressWasShown(value)
 
+method createMemberItem*[T](self: Module[T], memberId: string, state: MembershipRequestState, role: MemberRole): MemberItem =
```
Comment (jrainville) [inline src/app/modules/main/module.nim:1599]: Did you forget to create the forward declaration?

Comment (jrainville) [reply src/app/modules/main/module.nim:1599]: Maybe it's because of the `io_interface`?

Comment (mprakhov) [reply src/app/modules/main/module.nim:1599]: maybe... I don't have deep knowledge of NIM, but during dev testing, everything is ok + this should be a compilation error

Comment (jrainville) [reply src/app/modules/main/module.nim:1599]: Indeed. All's good then

File: src/app/modules/main/module.nim [hunk 3]
```diff
@@ -1649,4 +1596,23 @@ method checkIfAddressWasCopied*[T](self: Module[T], value: string) =
...
+    pubKey = memberId,
+    displayName = contactDetails.dto.displayName,
+    ensName = contactDetails.dto.name,
+    isEnsVerified = contactDetails.dto.ensVerified,
+    localNickname = contactDetails.dto.localNickname,
+    alias = contactDetails.dto.alias,
+    icon = contactDetails.icon,
+    colorId = contactDetails.colorId,
+    colorHash = contactDetails.colorHash,
+    onlineStatus = toOnlineStatus(self.controller.getStatusForContactWithId(memberId).statusType),
```
Comment (MishkaRogachev) [inline src/app/modules/main/module.nim:1611]: suggestion:  
```
let status = self.controller.getStatusForContactWithId(memberId)
...
onlineStatus = toOnlineStatus(status.statusType)
```

Comment (mprakhov) [reply src/app/modules/main/module.nim]: done

**Pull request merged**